File: Accounts/Pathlock/gen-2025-03-19/pages/login-page.ts
import { Page, Locator, expect } from '@playwright/test';

export class LoginPage {
  readonly page: Page;
  readonly usernameInput: Locator;
  readonly passwordInput: Locator;
  readonly loginButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.usernameInput = page.locator('#ctl00_ctl00_PageContent_PageContent_Login1_UserName');
    this.passwordInput = page.locator('#ctl00_ctl00_PageContent_PageContent_Login1_Password');
    this.loginButton = page.locator('#ctl00_ctl00_PageContent_PageContent_Login1_LoginLinkButton');
  }

  async goto() {
    await this.page.goto('/App/LoginPage.aspx');
    await expect(this.page).toHaveURL(/LoginPage\.aspx/);
  }

  async login(username: string, password: string) {
    await this.usernameInput.fill(username);
    await this.passwordInput.fill(password);
    await this.loginButton.click();

    // Wait for navigation to complete after login
    await this.page.waitForNavigation();
  }
}

File: Sri/boring-todo/generated-ui-tests/pages/login_page.py
from playwright.sync_api import Page

class LoginPage:
    def __init__(self, page: Page):
        self.page = page
        self.email_input = page.locator('[data-test-id="email-input"]')
        self.password_input = page.locator('[data-test-id="password-input"]')
        self.login_button = page.locator('[data-test-id="login-button"]')

    def navigate(self):
        self.page.goto("http://localhost:3001/login")

    def login(self, email, password):
        self.email_input.fill(email)
        self.password_input.fill(password)
        self.login_button.click()
